pkg/apis/config/defaults.go

```
@@ -71,62 +71,130 @@ func NewDefaultsConfigFromConfigMap(config *corev1.ConfigMap) (*Defaults, error)
+			}
+			return nsConfig.BrokerConfig, nil
+		}
+		if config, ok := nsConfig.BrokerClasses[brokerClassName]; ok && config != nil {
+			return config, nil
+		}
+	}
+
+	// Check cluster default configuration
+	return d.getClusterDefaultBrokerConfig(brokerClassName)
+}
+
+// getBrokerConfigForEmptyClassName returns the BrokerConfig for the given namespace when brokerClassName is empty.
+// It first checks the namespace specific configuration, then the cluster default configuration.
+func (d *Defaults) getBrokerConfigForEmptyClassName(ns string) (*BrokerConfig, error) {
+	// Check if namespace has a default broker class
+	if nsConfig, ok := d.NamespaceDefaultsConfig[ns]; ok && nsConfig != nil {
+		if nsConfig.DefaultBrokerClass != "" {
+			return d.getBrokerConfigByClassName(ns, nsConfig.DefaultBrokerClass)
+		}
+	}
+
+	// Fallback to cluster default configuration
+	return d.getClusterDefaultBrokerConfig("")
+}
+
+// getClusterDefaultBrokerConfig returns the BrokerConfig for the given brokerClassName.
+func (d *Defaults) getClusterDefaultBrokerConfig(brokerClassName string) (*BrokerConfig, error) {
+	if d.ClusterDefaultConfig == nil || d.ClusterDefaultConfig.BrokerConfig == nil {
+		return nil, errors.New("Defaults for Broker Configurations have not been set up.")
 	}
-	if d.ClusterDefault != nil && d.ClusterDefault.BrokerConfig != nil {
-		return d.ClusterDefault.BrokerConfig, nil
+
+	if brokerClassName == "" {
+		return d.ClusterDefaultConfig.BrokerConfig, nil
 	}
-	return nil, errors.New("Defaults for Broker Configurations have not been set up.")
+
+	// Check if the brokerClassName is the default broker class for the whole cluster
+	if d.ClusterDefaultConfig.DefaultBrokerClass == brokerClassName {
+		return d.ClusterDefaultConfig.BrokerConfig, nil
+	}
+
+	if config, ok := d.ClusterDefaultConfig.BrokerClasses[brokerClassName]; ok && config != nil {
+		return config, nil
+	}
+
+	return d.ClusterDefaultConfig.BrokerConfig, nil
```

Does it make sense to return an error here instead? At this point we know there is no config for the requested broker class, so we might get weird behaviour for brokers with a random config that doesn't match their class. @pierDipi @creydr any ideas?

---

Ping @Leo6Leo - wdyt here? From the previous comment, my intuition is that this should return an error, but I'm not sure